packages/gatsby-transformer-mdx-introspection/README.md

```diff
@@ -40,31 +47,122 @@ query MyQuery {
 }
 ```
 
-Please not the lowercase `apitype` in the query - the names are normalized to lowercase by the DOM parser used.
-
-Example response:
+### Example response:
 
 ```json
 {
   "data": {
     "allComponentInMdx": {
       "nodes": [
         {
+          "component": "ApiType",
           "attributes": [
             {
-              "name": "apikey",
+              "name": "apiKey",
               "value": "test"
             },
             {
               "name": "type",
               "value": "OutOfOrderPropertiesTestType"
             }
           ],
-          "page": {
-            "fields": {
-              "slug": "/api/types/"
+          "mdx": {
+            "file": {
+              "relativePath": "api/types/index.md"
             }
           }
         },
-...
+        ...
+      ]
+    }
+  }
+}
+```
+
+## Options
+
+**`lowercaseIdentifiers`** [boolean] (optional, defaults to false)
+
+Whether to have the `component` field (and `attribute.name`) be all lower case (legacy)
+
+**`trimWhitespace`** [boolean] (optional, defaults to true)
+
+Whether to trim leading/trailing whitespace in JSX snippets and string literals
+
+**`collapseWhitespace`** [boolean] (optional, defaults to true)
+
+Whether to collapse consecutive whitespace in JSX snippets and string literals
+
+**`attachAST`** [boolean] (optional, defaults to false)
```

Comment:
Just to be sure: Your text sounded like you aded this option for a potential user.  Was there an own use case in your site implementation for digging into the full AST and doing custom things?  

Not being able to trigger a potentially bad performing call is the best way to not have trouble with implementations that use it too liberally, hence I tend to not have it.

Reply:
I haven't actually gotten to implementing the algorithm I need for my own use case, but the main reason I added this was that I wanted to allow potential users to query the underlying Babel JSX AST. However, now that I consider it, since there is the `tree` field exposed for manual traversal of the reduced tree, I'm not very attached to keeping this option. The best option is probably to remove it for now and introduce it (or a more performant solution) later on if it is needed.